pywinauto/application.py

```diff
@@ -1247,6 +1248,33 @@ def kill(self):
     # Non PEP-8 aliases
     kill_ = Kill_ = kill
 
+    def is_process_running(self):
+        """
+        Checks that process is running.
+        Can be called before start/connect.
+
+        Returns True if process is running otherwise - False
+        """
+        is_running = False
+        try:
+            h_process = win32api.OpenProcess(
+                win32con.PROCESS_QUERY_INFORMATION,
+                0,
+                self.process)
+            is_running = win32process.GetExitCodeProcess(
+                h_process) == win32defines.PROCESS_STILL_ACTIVE
+        except (win32gui.error, TypeError):
+            is_running = False
+        return is_running
+
+    def wait_for_process_exit(self, timeout=10, retry_interval=1):

```

Comment:
Global timeout settings are kept in module `timings`. So I'd suggest to move default values to this module. Also I think too long `retry_interval` is not the best practice because flexible timing assumes it should finish much faster if the condition is already satisfied. So 1 sec. will look too long for user if process is finished after ~0.1 sec. Some reasonable `retry_interval` like 0.01 or 0.1 sec. will not eat too much CPU time but it will look more smoothly.